orchestrator/test_runner/src/erc_721_happy_path.rs

```diff
@@ -0,0 +1,638 @@
+        delay_for(Duration::from_secs(4)).await;
+        if Instant::now() - start > TOTAL_TIMEOUT {
+            panic!("Failed to update validator set");
+        }
+    }
+    assert!(starting_eth_valset_nonce != current_eth_valset_nonce);
+    info!("Validator set successfully updated!");
+}
+
+// Checks for a MsgValsetUpdatedClaim attestation where every validator is represented
+async fn check_valset_update_attestation(
+    grpc_client: &mut GravityQueryClient<Channel>,
+    keys: &[ValidatorKeys],
+) {
+    let mut found = true;
+    iterate_attestations(grpc_client, &mut |decoded: MsgValsetUpdatedClaim| {
+        // Check that each bridge validator is one of the addresses in our keys
+        for bridge_val in decoded.members {
+            let found_val = keys.iter().any(|key: &ValidatorKeys| {
+                let eth_pub_key = key.eth_key.to_address().to_string();
+                bridge_val.ethereum_address == eth_pub_key
+            });
+            if !found_val {
+                warn!(
+                    "Could not find BridgeValidator eth pub key {} in keys",
+                    bridge_val.ethereum_address
+                );
+            }
+            found &= found_val;
+        }
+    })
+    .await;
+    assert!(
+        found,
+        "Could not find the valset updated attestation we were looking for!"
+    );
+    info!("Found the expected MsgValsetUpdatedClaim attestation");
+}
+
+// Tests an ERC20 deposit and panics on failure
+#[allow(clippy::too_many_arguments)]
+pub async fn test_erc20_deposit_panic(
+    web30: &Web3,
+    contact: &Contact,
+    grpc_client: &mut GravityQueryClient<Channel>,
+    dest: CosmosAddress,
+    gravity_address: EthAddress,
+    erc20_address: EthAddress,
+    amount: Uint256,
+    timeout: Option<Duration>, // how long to wait for balance on cosmos to change
+    expected_change: Option<Uint256>, // provide an expected change when multiple transactions will take place at once
+) {
+    match test_erc20_deposit_result(
+        web30,
+        contact,
+        grpc_client,
+        dest,
+        gravity_address,
+        erc20_address,
+        amount,
+        timeout,
+        expected_change,
+    )
+    .await
+    {
+        Ok(_) => {
+            info!("Successfully bridged ERC20!")
+        }
+        Err(_) => {
+            panic!("Failed to bridge ERC20!")
+        }
+    }
+}
+
+/// this function tests Ethereum -> Cosmos deposits of ERC20 tokens
+/// it validates the that the contract being called is the gravity contract
+/// the erc20 provided is a valid erc20 implementation, sends the prescribed
+/// amount of tokens, and then finds the attestations submitted by orchestrators
+/// on the Cosmos chain. Finally it checks that the balance has changed by the
+/// correct amount, which is either the deposit amount or the 'expected change'
+/// provided by the caller
+#[allow(clippy::too_many_arguments)]
+pub async fn test_erc20_deposit_result(
```

Comment:
this needs to be renamed, also all the function arguments that include "erc20"

Reply:
Updated: 

```
pub async fn test_erc721_deposit_result(
    web30: &Web3,
    contact: &Contact,
    dest: CosmosAddress,
    gravity_address: EthAddress,
    gravityerc721_address: EthAddress,
    erc721_address: EthAddress,
    token_id: Uint256,
    timeout: Option<Duration>,
) -> Result<(), GravityError> {
```